### PR TITLE
Make CommunityCommons compatible with Mendix 10

### PR DESCRIPTION
**Pull request by lvanengelen**
- Update to Studio Pro 9.18.0 which is the earliest version that can be opened by Mendix 10
- Update runtime and test library dependencies
- Enable deprecation checking during the build
- Add `exportModule` Gradle task dependency on `copyRuntimeToUserlib` so local exports work by default.
- Some repository cleanup (committed files, ignored files)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Add reusable Java methods that can be called from microflows and custom Java act
 
 In version 7.2.0 of this module, we introduced a new way of dependency management using a Gradle build file. Unfortunately, this does not mean that obsolete JARs are automatically deleted from your apps' userlib folders when you import this module into your app model.
 
-The Community Commons container project contains a variety of predesigned unit tests. To use these tests, download the Unit Testing module from the Mendix Marketplace and run `gradle copyAllToUserlib`. This module has a dependency on the Object Handling module, so that module should also be imported to your app if you want to run the tests.
+The Community Commons container project contains a variety of predesigned unit tests. To use these tests, run `gradle copyAllToUserlib`.
```
Comment (atennapel) [inline README.md:7]: We could add "And then run the mpr in Studio Pro and run the tests in a web browser." or so.

File: src/CommunityCommons/javasource/myfirstmodule/actions/Test_Clone_Java.java [hunk 2]
```diff
@@ -29,6 +32,8 @@ public java.lang.Boolean executeAction() throws Exception
 		FileDocument doc = new FileDocument(getContext());
 		doc.setName("TestDocument");
 		doc.getMendixObject().setValue(getContext(), "__UUID__", UUID.randomUUID().toString());
+		InputStream is = new ByteArrayInputStream("foo".getBytes(StandardCharsets.UTF_8));
+		doc.getMendixObject().setValue(getContext(), "Contents", is);
```
Comment (jveldhuizen) [inline src/CommunityCommons/javasource/myfirstmodule/actions/Test_Clone_Java.java:36]: Unlike `__UUID__`, `Contents` is a public attribute. Can you do:
```suggestion
		doc.getMendixObject().setValue(getContext(), FileDocument.MemberNames.Contents.toString(), is);

```

File: .gitignore [hunk 3]
```diff
@@ -32,9 +32,14 @@ src/javasource/system/UserActionsRegistrar.java
 /build/
 private/cache/retriever/catalog.xml
 package.xml
-/src/CommunityCommons/userlib/
-/src/CommunityCommons/build/
 *.mda
+/src/CommunityCommons/build/
+/src/CommunityCommons/javasource/*/proxies/
```
Comment (jveldhuizen) [inline .gitignore:37]: Can you ignore `/src/CommunityCommons/javasource/system/` as well? Everything from the System module is generated, right?

File: src/CommunityCommons/javasource/myfirstmodule/actions/Test_Clone_Java.java (hunk 2 above)
Comment (lvanengelen) [reply src/CommunityCommons/javasource/myfirstmodule/actions/Test_Clone_Java.java:36]: Fixed

File: marketplace/release-notes/10.0.0.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Update to support Mendix Studio Pro version 10.
```
[the remark this answers is not in the document]
Comment (lvanengelen) [reply marketplace/release-notes/10.0.0.txt]: Done

[the remark this answers is not in the document]
Comment (lvanengelen) [reply marketplace/release-notes/10.0.0.txt]: Done

**Pull request merged**